Tests/tests.py

```diff
@@ -1,118 +1,239 @@
 import json
 import os
 import unittest
-from .. import app
+from flask import url_for
+from app import flask_app, db
 
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 
-class SetUpTestDatabase(unittest.TestCase):
+class BaseTestCase(unittest.TestCase):
     """ setting up the test database"""
     def setUp(self):
-        app.flask_app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///' + BASE_DIR + '/bucket_list_test.db'
-        app.db.session.close()
-        app.db.create_all()
-        self.client = app.flask_app.test_client()
+        flask_app.config.from_object('flask_settings.config_default.TestingConfig')
+        flask_app.config['SERVER_NAME'] = '127.0.0.0'
+        db.session.close()
+        db.create_all()
+        self.client = flask_app.test_client()
+
+        with flask_app.app_context():
+            # create a user
+            payload = {"first_name": "Sharon", "last_name": "Njihia",
+                       "email": "sharonkaren@gmail.com", "password": "password"}
+            self.client.post(url_for('user_registration'), data=json.dumps(payload),
+                             content_type='application/json')
+            # get auth token
+            payload = {"username": "sharonkaren@gmail.com", "password": "password"}
+            token = self.client.post("/auth", data=json.dumps(payload), content_type='application/json')
+            self.token = 'JWT ' + json.loads(token.get_data(as_text=True))['access_token']
 
 
 class TestRegistration(unittest.TestCase):
```

Comment:
Can this also inherit the `BaseTestCase`? So you don't have to repeat the `SetUp` stuff